Abdullah week3 JS homework

heartwhite opened the pull request:
!!! important..!!  
here there are some changes from old weeks . please look after "Week3/arrays.js"
 12 changes on head are old. (because I did my changes on pull requests page according to Marcianos feedback. afterwards I applied changes on my page on forked repo. that's why the files seem changed ... I'm sorry for the mess...

JUST klick "WEEK3"  (the link one before the last link) for HOMEWORK

heartwhite committed:
abdullah github-username:heartwhite

Javascript week1homework

heartwhite committed:
Merge pull request #1 from heartwhite/heartwhite-patch-1

abdullah github-username:heartwhite

heartwhite committed:
Update arrays.js

up to Marcianos feedback.  I used .push for adding

heartwhite committed:
Update checkType.js

compared all variables at all angles

heartwhite committed:
Update compairingInfinities.js

I took infinitives into Variables , than compared to make it more readable .

heartwhite committed:
Update helloInDifLang.js

console.log for each language

heartwhite committed:
Update numberRounding.js

changed calculating to Math.max to find highest.

heartwhite committed:
Update varXInteger.js

heartwhite committed:
Update arrays.js

.push chanced to myFavoriteAnimals variable

heartwhite committed:
Update MAKEME.md

heartwhite committed:
Update arrays.js

heartwhite committed:
Update checkType.js

heartwhite committed:
Update compairingInfinities.js

heartwhite committed:
Update helloInDifLang.js

heartwhite committed:
Update numberRounding.js

heartwhite committed:
Abdullah JS week 3

index.html and 19 js files for steps of homework

heartwhite committed:
Merge pull request #2 from heartwhite/heartwhite-patch-1

Heartwhite patch 1

heartwhite committed:
second console log added

i just now understand challenge. I added a second console log using '?'

marcianoviereck commented on Week1/numberRounding.js:
```diff
@@ -0,0 +1,15 @@
+let a = Math.round(z);
+console.log(a);
+let numbers = [a, z];
+let highNumber=(Math.max(z,a));
```
nice!

heartwhite replied:
thanks.

marcianoviereck commented on Week3/moreJs-10.js:
```diff
@@ -0,0 +1,26 @@
+    }
+    let lastTwoProduct = products[products.length - 2] + ' and' + products[products.length - 1] ;
+    console.log(lastTwoProduct);
+    products.splice((products.length - 2), 2,);
```
will this compile? with the last `,`?

heartwhite replied:
oh ! there is an extra ( unnecessary )  ' , '. 
I removed that. 
I don't know is it an error because it worked in console logs, but clearly it is weird.

marcianoviereck commented on Week3/moreJs-6.js:
```diff
@@ -0,0 +1,19 @@
+
+function vehicle (color, type, age){
+    let types = [' car' , ' motorbike'];
+    console.log('a ' + color +' ' + condition(age) + types[(type-1)]);
```
What happens if you pass '3' as value for the `type` as arguments? Will this still work?

marcianoviereck commented on Week3/moreJs-6.js:
```diff
@@ -0,0 +1,19 @@
+    let types = [' car' , ' motorbike'];
+    console.log('a ' + color +' ' + condition(age) + types[(type-1)]);
+}
+function condition(years){
```
Maybe the name `age` here is more consistent, because you use it aswell above

heartwhite replied:
yes. you re right. 
I was trying to find some other parameters name. I thought it might confuse ('age' in 2 places).
but I forgot about scopes, and parameters I think.
I changed it to age. its better

marcianoviereck commented on Week3/moreJs-9.js:
```diff
@@ -0,0 +1,20 @@
+    // and in case you can run in a html seperately.
+    console.log('a ' + color +' ' + condition(age) + vehicles[(type-1)]);
+}
+function condition(years){
```
Maybe the name age here is more consistent, because you use it aswell above

heartwhite replied:
yes. you re right. 
I was trying to find some other parameters name. I thought it might confuse ('age' in 2 places).
but I forgot about scopes, and parameters I think.
I changed it to age. its better

marcianoviereck commented on Week3/moreJs-9.js:
```diff
@@ -0,0 +1,20 @@
+*/
+
+function vehicle (color, type, age){
+    let vehicles = ['motorbike' , 'caravan', 'bike', 'car'];
```
What happens if you pass '10' as value for the type as arguments? Will this still work?

heartwhite replied:
it returns like ' undefined ' .
I added an if statement that logs a warning when the value is out of range ( length of array).

marcianoviereck reviewed (changes requested):
Overall the homework looks good! But there are just some remarks here and there, if you can fix those then the homework is approved!:)

heartwhite committed:
Update moreJs-6.js

1 - condition's (function) parameter changed as 'age' from 'years'. this way more consistent with lines above.
2- if statement added for the value of 'type' parameter. console logs the vehicle only if it has vehicle type. otherwise logs an warning

heartwhite committed:
Update moreJs-9.js

1 - condition's (function) parameter changed as 'age' from 'years'. this way more consistent with lines above.
2- if statement added for the value of 'type' parameter. console logs the vehicle only if it has vehicle type. otherwise logs an warning

heartwhite committed:
Update moreJs-10.js

removed unnecessary ',' from end of line 19.

marcianoviereck commented on Week3/moreJs-10.js:
```diff
@@ -0,0 +1,23 @@
+  products.splice(products.length - 2, 2);
+
+  console.log("Amazing Abdullah's Garage, we sell " + products.join() + ',' + lastTwoProduct + '.');
+}
```
This is fine!

Another way to do this is this:

```

vehicles = ["car", "bike", "train"];

function advertisement() {
  let fullString = "Amazing Joe's Garage, we service ";
  for (let i = 0; i < vehicles.length; i++) {
    const element = vehicles[i];
    if (i < vehicles.length - 2) {
      fullString += vehicles[i] + ', ';
    } else if (i === vehicles.length - 2) {
      fullString += vehicles[i];
    } else {
      fullString += ' and ' + vehicles[vehicles.length - 1] + "."
    }
  }
  console.log(fullString);
}
```

heartwhite replied:
I think your example automate better. Thanks. it will be in my codes today before evening

marcianoviereck reviewed (approved):
Nice! The homework is approved.
